Applications/spire/include/spire/charting/chart_view.hpp

```diff
@@ -25,6 +26,22 @@ namespace Spire {
         \param bottom_right The bottom right point to display.
       */
       void set_region(ChartPoint top_left, ChartPoint bottom_right);
+
+    protected:
+      void paintEvent(QPaintEvent* event) override;
+
+    private:
+      ChartValue::Type m_x_axis_type;
+      ChartValue::Type m_y_axis_type;
+      ChartPoint m_top_left;
+      ChartPoint m_bottom_right;
+      QFont m_label_font;
+      std::string m_timestamp_format;
+      CustomVariantItemDelegate* m_item_delegate;
+
+      QString drawable_timestamp(const boost::posix_time::ptime& time);
+      ChartValue get_step(const ChartValue::Type& value_type,
```

Comment:
Pass by value, mark the method as ```const```.

Reply:
I can't mark this constant because it modifies m_timestamp_format. Or should I make the timestamp member mutable?

Reply:
If that's the case then rename the function. A function named ```get_xxx``` must not modify any state.